pkg/sfu/audio/audiolevel.go

```diff
@@ -48,34 +58,44 @@ func (l *AudioLevel) Observe(level uint8, durationMs uint32) {
 
 	if level <= l.params.ActiveLevel {
 		l.activeDuration += durationMs
-		if l.observeLevel > level {
-			l.observeLevel = level
+		if l.loudestObservedLevel > level {
+			l.loudestObservedLevel = level
 		}
 	}
 
 	if l.observedDuration >= l.params.ObserveDuration {
 		// compute and reset
 		if l.activeDuration >= l.minActiveDuration {
-			level := uint32(l.observeLevel) - uint32(20*math.Log10(float64(l.activeDuration)/float64(l.params.ObserveDuration)))
-			l.currentLevel.Store(level)
+			// adjust loudest observed level by how much of the window was active.
+			// Weight will be 0 if active the entire duration
+			// > 0 if active for longer than observe duration
+			// < 0 if active for less than observe duration
+			activityWeight := 20 * math.Log10(float64(l.activeDuration)/float64(l.params.ObserveDuration))
+			adjustedLevel := float64(l.loudestObservedLevel) + activityWeight
+			linearLevel := ConvertAudioLevel(adjustedLevel)
+
+			// exponential smoothing to dampen transients
+			smoothedLevel := l.smoothedLevel.Load()
+			smoothedLevel += (linearLevel - smoothedLevel) * l.smoothFactor
+			smoothedLevel = math.Ceil(smoothedLevel*audioLevelQuantization) * invAudioLevelQuantization
```

Comment:
Have no idea what this quantisation is trying to achieve, but this does affect things a bit. Previously, sorting was done before quantisation. Now, it is done after.

Reply:
Okay, have removed this quantisation as the purpose of that is not clear to me. Can re-introduce if it does serve a real purpose.